### PR TITLE
Small changes to help enable subcell for Newtonian Euler

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Various changes to code in Evolution that are helpful when enabling subcell for the Newtonian Euler system, plus some cleanup.

- Convert test Burgers.Flux to pypp test
- Remove evolution::dg::ConservativeDuDt
- Remove unused type aliases from conservative systems
- Fix instantiations for NewtoniainEuler systems with non-zero source
- Change implementation of source terms for IsentropicVortex

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/NewtonianEuler/Sources/VortexPerturbation.hpp [hunk 1]
```diff
@@ -78,28 +77,20 @@ struct VortexPerturbation {
...
-             gsl::not_null<tnsr::I<DataVector, Dim>*> source_momentum_density,
-             gsl::not_null<Scalar<DataVector>*> source_energy_density,
-             const NewtonianEuler::Solutions::IsentropicVortex<Dim>& vortex,
-             const tnsr::I<DataVector, Dim>& x, double time) const;
+  static void apply(
+      gsl::not_null<Scalar<DataVector>*> source_mass_density_cons,
+      gsl::not_null<tnsr::I<DataVector, 3>*> source_momentum_density,
+      gsl::not_null<Scalar<DataVector>*> source_energy_density,
+      const NewtonianEuler::Solutions::IsentropicVortex<3>& vortex,
+      const tnsr::I<DataVector, 3>& x, double time);
```
Comment (nilsdeppe) [inline src/Evolution/Systems/NewtonianEuler/Sources/VortexPerturbation.hpp:93]: why is this function no longer `const`?

**Review by nilsdeppe — APPROVED** [review]
LGTM, just one question. If you end up changing anything, please squash immediately :)

File: src/Evolution/Systems/NewtonianEuler/Sources/VortexPerturbation.hpp (hunk 1 above)
Comment (kidder) [reply src/Evolution/Systems/NewtonianEuler/Sources/VortexPerturbation.hpp:93]: static functions cannot be const (clang-tidy wants it to be a static function...)

Comment (nilsdeppe) [reply src/Evolution/Systems/NewtonianEuler/Sources/VortexPerturbation.hpp:93]: ah! That's what I missed... Thanks :)

**Pull request merged**